dev-db/db/db18-18.1.40.recipe

```diff
@@ -65,36 +62,66 @@ PROVIDES_devel="
 	devel:libdb_18.1$secondaryArchSuffix = $portVersion
 	devel:libdb_cxx$secondaryArchSuffix = $portVersion
 	devel:libdb_cxx_18.1$secondaryArchSuffix = $portVersion
-#	devel:libdb_java$secondaryArchSuffix = $portVersion
-#	devel:libdb_java_18.1$secondaryArchSuffix = $portVersion
 	"
 REQUIRES_devel="
 	db18$secondaryArchSuffix == $portVersion base
 	"
 
 BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
-#	openjdk17$secondaryArchSuffix
 	devel:libcrypto$secondaryArchSuffix
 	devel:libssl$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
 	haiku${secondaryArchSuffix}_devel
 	cmd:gcc$secondaryArchSuffix
-#	cmd:javac >= 17
 	cmd:ld$secondaryArchSuffix
 	cmd:libtool
 	cmd:make
 	cmd:pkg_config$secondaryArchSuffix
 	"
 
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+	PROVIDES_devel+="
+		devel:libdb_java$secondaryArchSuffix = $portVersion
+		devel:libdb_java_18.1$secondaryArchSuffix = $portVersion
+		"
+	REQUIRES_devel+="
+		db18${secondaryArchSuffix}_java == $portVersion
+		"
+
+	SUMMARY_java="Java bindings for the Berkeley Database"
+	PROVIDES_java="
+		db18${secondaryArchSuffix}_java = $portVersion
+		lib:libdb_java_18$secondaryArchSuffix = $portVersion
+		lib:libdb_java_18.1$secondaryArchSuffix = $portVersion
+		lib:libdb_java_18.1_g$secondaryArchSuffix = $portVersion
+		"
+	REQUIRES_java="
+		db18$secondaryArchSuffix == $portVersion base
+		$REQUIRES
+		java:runtime == 17
+		"
+
+	BUILD_REQUIRES+="
+		java:environment == 17
+		"
+	BUILD_PREREQUIRES+="
+		cmd:javac >= 17
+		"
+fi
+
 BUILD()
 {
-#	export CPPFLAGS="-I/boot/system/lib/openjdk17/include/haiku"
+	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+		source /system/data/profile.d/openjdk.sh
+		export CPPFLAGS="-I/boot/system/lib/openjdk17/include/haiku"
+		java="--enable-gui --enable-java"
```

Comment:
You could leave out the GUI and provide just the Java bindings part.

Reply:
It was worth a try :) Will do some check tomorrow on 32bit for this. Thanks so far!